### PR TITLE
Finish User Gateway Factory Implementation for Log in

### PR DESCRIPTION
**Pull request by KaxiYuan8288**

### CODE REVIEW

File: src/main/java/gateway/UserGatewayFactory.java
Comment (Matt-1369) [inline src/main/java/gateway/UserGatewayFactory.java:1]: The method "isClient" should throw FailMessageError.

Comment (Matt-1369) [inline src/main/java/gateway/UserGatewayFactory.java:1]: UserGatewayFactory either needs to handle FailMessageError or throw a new Error.

File: src/main/java/userlogin/UserLoginInteractor.java
Comment (Matt-1369) [inline src/main/java/userlogin/UserLoginInteractor.java:1]: In the login method, it is necessary to handle the error generated by CreateUserGateway.

File: src/main/java/screen/FailMessage.java
Comment (Matt-1369) [inline src/main/java/screen/FailMessage.java:1]: The error name "FaillMessage" is not clear, it can be changed to "ApplicationException".

**Review by josephpcsywm — APPROVED** [review]
UserGatewayFactory correctly added and use login use case is fixed

**Pull request merged**